test/functional/smart_answers_controller_test.rb

```diff
@@ -193,6 +194,54 @@ def teardown
         assert_select "pre.debug", false, "The page should not render debug information"
       end
     end
+
+    context "A/B testing" do
+      setup do
+        # Switch on feature flag
+        ENV['ENABLE_NEW_NAVIGATION'] = 'yes'
```

Comment:
Unless ENV is being reset somewhere this is leaking state. Probably best to stub access to that key.